peeringdb/filters.py

```diff
@@ -23,15 +23,17 @@ class NetworkIXLanFilterSet(django_filters.FilterSet):
 
     class Meta:
         model = NetworkIXLan
-        fields = ["id", "asn"]
+        fields = ["id", "asn", "net__policy_general"]
 
     def search(self, queryset, name, value):
         if not value.strip():
             return queryset
         try:
-            qs_filter = Q(asn=int(value.strip()))
+            qs_filter = Q(asn__icontains=int(value.strip())) | Q(
+                net__name__icontains=value
+            )
         except ValueError:
-            return queryset
+            qs_filter = Q(net__name__icontains=value)
```

Comment:
I would probably do something like that instead, to avoid duplicating `Q()` filters.

```python
qs_filter = Q(net__name__icontains=value)
try:
    qs_filter |= Q(asn=int(value.strip()))
except ValueError:
    pass
```

Reply:
Good idea.